daceml/onnx/op_implementations/fpga_implementations.py

```diff
@@ -886,6 +886,1132 @@ def make_compute(sdfg, state, vec_width=1):
         return new_sdfg
 
 
+@op_implementation(op="Conv", name="fpga_tiled")
+class FPGAIm2ColConv_tiled(ONNXForward):
+    # ==================================
+    # Im2Col Convolution based on DaCe Master GEMM
+    # ==================================
+    '''
+    Use tiled 1D-Systolic GEMM implementation Y = W @ X + B (A @ B + C)
+    A: the weights W streamed for Im2Col processing
+    B: the Im2Col converted input X
+    C: the Bias B (if applicable)
+
+    Let:        
+    K = num_channels * filter_hx * filter_hy
+    M = output_size_y * output_size_x
+    N = num_filters
+
+
+    General outline of the implementation:
+    The convolution is performed using the Im2Col approach i.e. by performing a GEMM
+    on a matrix constructed from the input data, which is stored in
+    (Batch, Channel, X, Y) layout. As the Im2Col matrix contains a lot of duplicate values,
+    we do not materialize it in memory, but read from the normal data layout in memory and
+    stream the virtual matrix to the systolic array. The weights W matrix i.e. the A in the GEMM
+    has dimensions NxK and the data matrix X i.e. the B in the GEMM has layout KxM, which results
+    in a matrix of dimensions NxM.
+
+    To scale the convolution up to high-resolution image data, we use tiling. The tiling happens
+    on the virtual Im2Col matrix, thus all loops are with respect to the GEMM computation, to be
+    able to tile on the GEMM itself. Having the loop bounds w.r.t to the GEMM requires index
+    translation, when reading data (weights and image in `make_read_W` and `make_read_X`) from DRAM, because
+    the GEMM accesses the Im2Col matrix, but the data layout is (b, c, x, y).
+
+    Tiling happens along two dimensions. Along the N dimension, which is computed in parallel by the systolic
+    array and there will be `ceil(N / P)` many tiles. The second tiling happens along the dimension M, which is
+    the product of the output image dimensions. To simplify the tiling scheme implementation only multiples of the
+    image width are allowed as a tile size (but it does not need to evenly divide the whole image).
+    '''
+    @staticmethod
+    def forward_can_be_applied(node: ONNXOp, state: SDFGState,
+                               sdfg: SDFG) -> bool:
+        X = in_desc_with_name(node, state, sdfg, "X")
+        W = in_desc_with_name(node, state, sdfg, "W")
+        Y = out_desc_with_name(node, state, sdfg, "Y")
+
+        try:
+            B = in_desc_with_name(node, state, sdfg, "B")
+        except Exception as e:
```

Comment:
Too broad, catch one or more specific exceptions and write why it's fine to catch them

Reply:
addressed